Sources/Fuzzilli/Core/ProgramBuilder.swift

```diff
@@ -299,41 +303,125 @@ public class ProgramBuilder {
                 parameterTypes.append(varargsParam.removingFlagTypes())
             }
         }
-            
+
         for param in parameterTypes {
             if param.isOptional {
                 // It's an optional argument, so stop here in some cases
                 if probability(0.25) {
                     break
                 }
             }
-            
+
             assert(!param.isList)
-            guard let v = randVar(ofType: param) else { return nil }
-            arguments.append(v)
+            if let v = randVar(ofType: param) {
+                arguments.append(v)
+            } else if withGeneration {
+                let argument = generateVariable(ofType: param)
+                assert(randVar(ofType: param) != nil)
+                arguments.append(argument)
+            } else {
+                return nil
+            }
         }
-            
+
         return arguments
     }
-    
+
     public func generateCallArguments(for function: Variable) -> [Variable]? {
         let signature = type(of: function).signature ?? FunctionSignature.forUnknownFunction
         return generateCallArguments(for: signature)
     }
-    
+
     public func generateCallArguments(forMethod methodName: String, on object: Variable) -> [Variable]? {
         let signature = methodSignature(of: methodName, on: object)
         return generateCallArguments(for: signature)
     }
-    
-    
+
+    /// Generates a sequence of instructions that generate the desired type.
+    func generateVariable(ofType type: Type) -> Variable {
+        switch type.baseType {
```

Comment:
Maybe we can structure this a little different:

- First, a number of checks of this form:
    `if type.Is(.integer) || type.Is(fuzzer.environment. intType { // do the loadInt/loadFloat/createObject/... instruction directly` for all types here: https://github.com/googleprojectzero/fuzzilli/blob/master/Sources/Fuzzilli/Core/Environment.swift#L48
- assert that the type is now `.object` (I think this must be the case now)
- check if a group is present, if yes do the necessary thing
- construct a plain object with properties and methods, as you already do

I think this would get rid of some nesting levels and make the code easier to follow